apps/api/src/modules/summary/summary.service.ts

```diff
@@ -46,27 +49,87 @@ export class SummaryService extends CoreService<Summary> {
           .on('error', (error) => {
             reject(error);
           });
-      });
+        });
+      }}
+    
+    getSummaryById(jobId: number): Promise<Summary[]> {
+        this.logger.log(`Getting summary with id: ${jobId}`);
+        return this.summaryRepository.find({
+          where: {
+            jobId: jobId,
+            status: Status.ACTIVE,
+          },
+        });
     }
-  }
 
-  async findAllJobs(
-    options: QueryOptionsDto): Promise<SummaryResponse> {
-    this.logger.log('Getting all jobs with options.');
 
-    const baseConditions = [
-      { field: 'status', value: Status.ACTIVE },
-    ];
-    const searchableFields = ['createdBy', 'data', 'result'];
 
-    const { items, total } = await super.findAll(
-      options,
-      'summary',
-      searchableFields,
-      baseConditions,
-    );
-    return new SummaryResponse(items, total, options.offset, options.limit);
-  }
+    getPendingSummary(): Promise<Summary[]> {
+        this.logger.log('Getting all active pending summaries');
+        return this.summaryRepository.find({
+          where: {
+            jobStatus: JobStatus.PENDING,
+            status: Status.ACTIVE,
+          },
+        });
+      }
 
-}
 
+    async addSummaryToQueue(): Promise<void> {
+        this.logger.log('Starting CRON job to add pending summaries to queue');
+    
+        if (this.isProcessingQueue) {
+          this.logger.log('Summaries are already being added to queue, skipping this CRON job');
+          return;
+        }
+    
+        this.isProcessingQueue = true;
+        let allPendingSummaries = [];
+    
+        try {
+          allPendingSummaries = await this.getPendingSummary();
+        } catch (error) {
+          this.isProcessingQueue = false;
+          this.logger.error('Error fetching pending summaries');
+          this.logger.error(JSON.stringify(error, null, 2));
+          return;
+        }
+    
+        this.logger.log(`Adding ${allPendingSummaries.length} pending summaries to queue`);
+    
+        for (const summary of allPendingSummaries) {
+          try {
+            summary.jobStatus = JobStatus.IN_PROGRESS;
+            await this.summaryQueueService.addsummaryToQueue(summary);
+          } catch (error) {
+            summary.jobStatus = JobStatus.PENDING;
+            summary.result = { result: error };
+            this.logger.error(`Error adding summary with id: ${summary.id} to queue`);
+            this.logger.error(JSON.stringify(error, null, 2));
+          } finally {
+            await this.summaryRepository.save(summary);
+          }
+        }
+    
+        this.isProcessingQueue = false;
+      }
+    
+
+      async findAllJobs(
```

Comment:
findAllJobs is already added. Check your code properly again, I think its not merged properly with main. Or did you copy paste?
Try to copy the main branch code and then add your code after that so that you don't disturb the code.

Reply:
Copied code from main branch and then added my code